fronted/src/components/PopupWithForm.js

```diff
@@ -1,12 +1,32 @@
 import React from "react";
 import './PopupWithForm.css'
+import FormValidator from "../utils/Validation";
 
 function PopupWithForm(props) {
+  const formRef = React.useRef();
+
+
+  React.useEffect(() => {
+    if (props.isValidatedForm) {
+      const settings = {
+        inputSelector: ".popup__form-input",
+        submitButtonSelector: ".popup__save-button",
+        inactiveButtonClass: "popup__save-button_disabled",
+        inputErrorClass: "popup__form-input_type_error",
+        errorClass: "popup__form-input-error_active",
+      };
+      const validatedForm = new FormValidator(settings, formRef.current);
+      validatedForm.enableValidation();
+    }
+  }, [props.isValidatedForm]);
+
+
+
```

Comment:
**Might be better**


If it’s interesting for you here is a custom `validation hook`.

You should put it in a separate file  `useFormAndValidation.js` in folder `hooks` and then import it to the component that needs validation:

```js

import {useState, useCallback} from 'react';

export function useFormAndValidation() {
  const [ values, setValues ] = useState({});
  const [ errors, setErrors ] = useState({});
  const [ isValid, setIsValid ] = useState(true);

  const handleChange = (e) => {
    const {name, value} = e.target
    setValues({...values, [name]: value });
    setErrors({...errors, [name]: e.target.validationMessage});
    setIsValid(e.target.closest('form').checkValidity());
  };



  const resetForm = useCallback((newValues = {}, newErrors = {}, newIsValid = false) => {
    setValues(newValues);
    setErrors(newErrors);
    setIsValid(newIsValid);
  }, [setValues, setErrors, setIsValid]);

  return { values, handleChange, errors, isValid, resetForm, setValues, setIsValid };
}


```

And only one line of code enables validation:

```js

  const {values, handleChange, errors, isValid, setValues, resetForm} = useFormAndValidation()

```

So, you don’t need to make functions for handling inputs, storing values, resetting the form. Everything is already done.